Core/src/main/java/org/gusdb/fgputil/StringUtil.java

```diff
@@ -71,4 +71,75 @@ public static String rtrim(final String target, final char trim) {
   public static String trim(final String target, final char trim) {
     return ltrim(rtrim(target, trim), trim);
   }
+
+  /**
+   * Tests whether the given string is in the valid UUID V4 format.
+   *
+   * @param s String to test.
+   *
+   * @return {@code true} if the given string is a valid UUID V4 string,
+   * otherwise {@code false}.
+   */
+  // XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
```

Comment:
Should probably include a link to [the RFC](https://www.ietf.org/rfc/rfc4122.txt), but curious why you didn't just use regex here.  Might be marginally slower but much more clear and [this stackoverflow](https://stackoverflow.com/questions/7905929/how-to-test-valid-uuid-guid/13653180) shows that there are a few subtle requirements that your code below does not cover (first answer covers all versions, second answer differentiates).  Version 4 regex (I think this is javascript but could be converted to Java) is:
`/^[0-9A-F]{8}-[0-9A-F]{4}-[4][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i`

Reply:
I intentionally avoided regex since this will be called repeatedly, but if you think that's better I will replace it.